### PR TITLE
Add optional 'style' parameter to geom_violin

### PR DESCRIPTION
**Pull request by Fischmiep**
This patch solves #369. It adds an optional 'flat' parameter to geom_violin which allows drawing of 'flat-violin' or 'half-violin' plots. This effectively removes half of the violin, drawing the density only on the other side of the sample position. This opens up the left half for additional plots.

The "flat" side may be defined by an additional 'flat_side' parameter

Inspired by https://gist.github.com/dgrtwo/eb7750e74997891d7c20

### CODE REVIEW

**Review by has2k1 — COMMENTED** [review]
Good feature. The reference test image has the wrong size. How are you generating it? It should be done from the command line.

File: plotnine/geoms/geom_violin.py [hunk 1]
```diff
@@ -19,15 +19,23 @@ class geom_violin(geom):
...
+    draw_quantiles : float or [float]
        draw horizontal lines at the given quantiles (0..1)
        of the density estimate.
+    flat : :py:`bool`, optional (default: :py:`False`)
+       If :py:`False`, draw a regular violin plot. If :py:`True`,
+       draw a 'flat-violin' or 'half-violin' plot.
+    flat_side : :py:`str`, optional (default: :py:`'left'`)
+        If :py:`flat` is set: defines the "flat" side of the violin,
+        i.e. the side that is cut off. May be either :py:`'right'`
+        or :py:`'left'`
```
[the remark this answers is not in the document]
Comment (Fischmiep) [reply plotnine/geoms/geom_violin.py:31]: I wasn't quite satisfied with the two parameter solution either. `style` sounds good to me; I think it grasps the scenario pretty well.

**Comment by Fischmiep** [comment]
I actually created it using the command line but for some reason default images seem to be 480x640 instead of 430x559 for me. I'll have a look once I have some time tinkering with it.

**Comment by Fischmiep** [comment]
I managed to set the correct image dimensions but the image created by the test seems to diff. I guess due to different dpi settings? I'll have another look once I get around to it but any pointers/ideas for easy solutions would be appreciated.

**Comment by has2k1** [comment]
Try

```bash
export MPLLOCALFREETYPE=1
pip install matplotlib --no-binary matplotlib  # You may need to uninstall the previous one.
pytest -k style
```

Then copy the result image.


Also rename the test functions to `test_style`.

**Comment by Fischmiep** [comment]
There we go, reinstalling all dependencies seemed to have done the trick. Not sure what was going on there but whatever.

**Comment by has2k1** [comment]
@Fischmiep thanks a lot.

**Pull request merged**